### PR TITLE
Fk on update delete actions

### PR DESCRIPTION
**Pull request by mozarcik**
Not sure if i placed those correctly. I thought first to add those actions to ForeignKeyReference but there can be many references and action is defined once for fk. I also don't know if adding method to append those actions to base QueryBuilder class is proper.

Let me know if those should be changed.

### CODE REVIEW

**Comment by agrosner** [comment]
@mozarcik hey can you make this against the ```develop``` branch? This would be great to add into 1.2.0 and that's the current branch for the update.

File: flowcore/src/main/java/com/raizlabs/android/dbflow/sql/QueryBuilder.java [hunk 1]
```diff
@@ -164,6 +167,34 @@ public QueryClass appendNotEmpty(String string) {
         return castThis();
     }
 
+    /**
+     * Appends ON DELETE or ON UPDATE Action defined in ForeignKeyReference.
+     *
+     * @param action defined {@link com.raizlabs.android.dbflow.annotation.Action}
+     * @return QueryBuilder
+     */
+    public QueryBuilder appendAction(Action action) {
```
Comment (agrosner) [inline flowcore/src/main/java/com/raizlabs/android/dbflow/sql/QueryBuilder.java]: I would do something like ```action.name().replace("-"," ")``` instead to keep it simple and it does pretty much the same as here. Also I would add it as a method in the ```TableCreationQueryBuilder``` instead and use it vs. a ```QueryBuilder``` in this scenario.

File: compiler/src/main/java/com/raizlabs/android/dbflow/processor/writer/CreationQueryWriter.java [hunk 2]
```diff
@@ -121,7 +121,12 @@ public void write(JavaWriter javaWriter) throws IOException {
                         }
 
                         foreignKeyQueryBuilder.appendArray(columns)
-                                .append(")").appendSpaceSeparated("REFERENCES %1s");
+                                .append(")").appendSpaceSeparated("REFERENCES %1s")
+                                .append("ON UPDATE")
```
Comment (agrosner) [inline compiler/src/main/java/com/raizlabs/android/dbflow/processor/writer/CreationQueryWriter.java]: ```appendSpaceSeparated()``` here

File: compiler/src/main/java/com/raizlabs/android/dbflow/processor/writer/CreationQueryWriter.java [hunk 3]
```diff
@@ -121,7 +121,12 @@ public void write(JavaWriter javaWriter) throws IOException {
...
+                                .appendAction(foreignKeyField.column.onUpdate())
+                                .appendSpace()
```
Comment (agrosner) [inline compiler/src/main/java/com/raizlabs/android/dbflow/processor/writer/CreationQueryWriter.java]: use ```appendSpaceSeparated("ON DELETE")``` here instead

File: flowcore/src/main/java/com/raizlabs/android/dbflow/annotation/Action.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.raizlabs.android.dbflow.annotation;
+
+/**
+ * Actions associated with on update and on delete
+ */
+public enum Action {
```
Comment (agrosner) [inline flowcore/src/main/java/com/raizlabs/android/dbflow/annotation/Action.java]: Not sure if the name is descriptive enough. I would either do ```FKAction``` or ```ForeignKeyAction``` (I like this one).

**Comment by agrosner** [comment]
@mozarcik Other than these comments, I will merge these in when I merge develop. I may just cherry pick your commits instead.

**Comment by mozarcik** [comment]
Opened new PR based on develop branch (#26) so closing this one.

**Pull request closed without merging**